.github/eks-workflow-files/jio-eks/jio-template.yaml

```diff
@@ -0,0 +1,549 @@
+        completions: 1
+        parallelism: 1
+        template:
+          metadata:
+            labels:
+              node-role: gateway
+          spec:
+            # NO hostNetwork for gateway
+            tolerations:
+              - key: nvidia.com/gpu
+                operator: Exists
+                effect: NoSchedule
+            containers:
+            - name: gateway-container
+              image: PLACEHOLDER # HERE WE NEED JAX_VLLM_OFFLOADING_IMAGE
+              command:
+              - bash
+              - -c
+              - |
+                set -euo pipefail
+
+                echo "=== Starting Gateway ==="
+                echo "Gateway will listen on 0.0.0.0:50051"
+
+                cd /opt/jtbx/jax-inference-offloading
+                python -u jax_inference_offloading/controller/gateway.py 2>&1 | tee gateway.log
+
+              env:
+              - name: GATEWAY_PORT
+                value: "50051"
+
+              ports:
+              - containerPort: 50051
+                name: gateway
+                protocol: TCP
+
+              resources:
+                requests:
+                  cpu: "8"
+                  memory: "16Gi"
+                limits:
+                  cpu: "8"
+                  memory: "16Gi"
+
+              volumeMounts:
+              - name: shmem
+                mountPath: /dev/shm
+
+            volumes:
+            - name: shmem
+              emptyDir:
+                medium: Memory
+                sizeLimit: 16Gi
+
+  # vLLM Node (Node 0)
+  - name: vllm-node
+    replicas: 1
+    template:
+      spec:
+        backoffLimit: 0
+        completionMode: Indexed
+        completions: 1
+        parallelism: 1
+        template:
+          metadata:
+            labels:
+              node-role: vllm
+          spec:
+            serviceAccountName: jax-worker
+            hostNetwork: true
+            dnsPolicy: ClusterFirstWithHostNet
+            # avoid being in the same node as jax
+            affinity:
+              podAntiAffinity:
+                requiredDuringSchedulingIgnoredDuringExecution:
+                - labelSelector:
+                    matchExpressions:
+                    - key: node-role
+                      operator: In
+                      values:
+                      - jax
+                  topologyKey: kubernetes.io/hostname
+
+            containers:
+            - name: vllm-container
+              image: PLACEHOLDER # here we need JAX_VLLM_OFFLOADING_IMAGE
+              command:
+              - bash
+              - -c
+              - |
+                set -euo pipefail
+
+                GATEWAY_HOST="${JOBSET_NAME}-gateway-0-0.${JOBSET_NAME}"
+                export GATEWAY_URL="${GATEWAY_HOST}:50051"
+                HOST_IP=$(hostname -i) #-I | awk '{print $1}')
+                export RAY_HEAD_IP=${HOST_IP}
+                export JAX_COORDINATOR_ADDRESS="${JAX_COORDINATOR_HOST}:${JAX_COORDINATOR_PORT}"
```

Comment:
We don't need this for the vLLM container. This is only for the JAX processes to initialize themselves.